### PR TITLE
Lab2

### PR DESCRIPTION
**Pull request by mnmason86**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -98,6 +128,11 @@ function onClick(event){
     shownProducts[`${id[5]}`].imgTimesClicked++;
     generateAndRenderImage();
   }
+  if(localStorage.getItem('productData')){
+    OddDuckProduct.prototype.loadPriorData();
```
Comment (Barnord) [inline app.js:132]: Methods need to be called on specific objects, when you call it here your code has no idea what "this" is in the code block.

Additionally, this shouldn't be in an event listener. You only need to get your data once, and you only need to save it once.

File: app.js [hunk 2]
```diff
@@ -40,13 +30,52 @@ let allProducts = [
...
+  savedProducts.push(this);
+
+  for(let i = 0; i < allProducts.length; i++){
+    let product = allProducts[i];
+    savedProducts.push(product);
+  }
+  savedProducts = JSON.stringify(savedProducts);
+  localStorage.setItem('productData', savedProducts);
+
+};
```
Comment (Barnord) [inline app.js:71]: This method continually adds data to productData, but it never overwrites anything.

**Review by Barnord — COMMENTED** [review]
Great work, I left a couple notes that should hopefully help you finish this lab.

File: app.js (hunk 1 above)
Comment (mnmason86) [reply app.js:132]: Ok. David and Adam were under the impression that data needed to be saved on every click, this is why I was calling my save functions in the click handler. Thank you!

**Pull request merged**